app/lua/lgc.c

```diff
@@ -538,7 +555,7 @@ static void atomic (lua_State *L) {
   size_t udsize;  /* total size of userdata to be finalized */
   /* remark occasional upvalues of (maybe) dead threads */
   remarkupvals(g);
-  /* traverse objects cautch by write barrier and by 'remarkupvals' */
+  /* traverse objects caucht by write barrier and by 'remarkupvals' */
```

Comment:
I think you mean "caught" :rofl:

Reply:
A bit naff replacing one spelling error by another !!

Reply:
@jmattsson, re your previous example, this is a mindfart on your part. :grinning: It does exactly what I would expect it to do, and would have the same result as `wtf=loadfile('wtf.lua')` is this was in the SPIFFS.  Do a `luac.cross -l -o /dev/null local/lus/wtf.lua` to see what this generate
-  The first execution executes the following with replaces the global `wtf` with the closure:
```
	1	[4]	CLOSURE  	0 0	; 0x258b2d0
	2	[1]	SETGLOBAL	0 -1	; wtf
	3	[4]	RETURN   	0 1
```
-  `wtf` now points to the closed function so the second call execute the following giving the response you expected.
```
	1	[2]	GETGLOBAL	1 -1	; print
	2	[2]	MOVE     	2 0
	3	[2]	CALL     	1 2 1
	4	[3]	LOADK    	1 -2	; 1
	5	[3]	RETURN   	1 2
```

Reply:
Umm.. what? I'm not reassigning wtf between the invocations. I'm assigning the global `wtf` exactly once, then calling it twice. I sure as sundown don't expect my `wtf` variable/function to be reassigned just because I invoked it (given I don't reassign it in the function). Kindly explain my supposed mindfart further.

Reply:
No, you are getting caucht (joke) is the subtleties of Lua.  If you've got 15 mins, then just Skupe me or ping me an email and I'll Skype you.

Reply:
As Terry promised me: ./facepalm
Okay, the penny dropped:`function wtf() end` is of course the same as `wtf = function() end`, and the initial `wtf` was the file/module closure itself.